src/values/gradient.rs

```diff
@@ -530,6 +540,52 @@ impl LineDirection {
   }
 }
 
+/// Converts a standard gradient direction to its legacy vendor-prefixed form.
+///
+/// Inverts keyword-based directions (e.g., `to bottom` → `top`) for compatibility
+/// with legacy prefixed syntaxes.
+///
+/// See: https://github.com/parcel-bundler/lightningcss/issues/918
+fn convert_to_legacy_direction(direction: &LineDirection) -> LineDirection {
+  match direction {
+    LineDirection::Horizontal(HorizontalPositionKeyword::Left) => {
+      LineDirection::Horizontal(HorizontalPositionKeyword::Right)
+    }
+    LineDirection::Horizontal(HorizontalPositionKeyword::Right) => {
+      LineDirection::Horizontal(HorizontalPositionKeyword::Left)
+    }
+    LineDirection::Vertical(VerticalPositionKeyword::Top) => {
+      LineDirection::Vertical(VerticalPositionKeyword::Bottom)
+    }
+    LineDirection::Vertical(VerticalPositionKeyword::Bottom) => {
+      LineDirection::Vertical(VerticalPositionKeyword::Top)
+    }
+    LineDirection::Corner { horizontal, vertical } => LineDirection::Corner {
+      horizontal: match horizontal {
+        HorizontalPositionKeyword::Left => HorizontalPositionKeyword::Right,
+        HorizontalPositionKeyword::Right => HorizontalPositionKeyword::Left,
+      },
+      vertical: match vertical {
+        VerticalPositionKeyword::Top => VerticalPositionKeyword::Bottom,
+        VerticalPositionKeyword::Bottom => VerticalPositionKeyword::Top,
+      },
+    },
+    LineDirection::Angle(angle) => {
+      let angle = angle.clone();
+      match angle {
+        Angle::Deg(number) => {
+          // Add 90 degrees
+          let number = (450.0 - number).abs() % 360.0;
+          // Round the number to 3 decimal places
+          let number = (number * 1000.0).round() / 1000.0;
+          LineDirection::Angle(Angle::Deg(number))
+        }
```

Comment:
This conversion is ported from [Autoprefixer’s implementation](https://github.com/postcss/autoprefixer/blob/main/lib/hacks/gradient.js#L103-L112)

Reply:
The link here requires the use of “Copy permalink”

![image](https://github.com/user-attachments/assets/cb9e6972-b9e4-42b7-a148-526a1610cbdf)